scripts/wawCommons.py

```diff
@@ -526,4 +526,85 @@ def replaceValue(sourceJson, target, replacementJson, matchKey = True):
+    logger.info("Calling function url '%s'", functionCallUrl)
+
+    functionResponse = requests.post(functionCallUrl, auth=(username, password),
+                                 headers={'Content-Type': 'application/json',
+                                          'accept': 'application/json'},
+                                 data=data)
+
+    if functionResponse.status_code == 200:
+
+        responseContentType = functionResponse.headers.get('content-type')
+        if responseContentType != 'application/json':
+            logger.error('Response content type is not json, content type: %s, response:\n%s', responseContentType, functionResponse.text)
+            return None
+
+        return functionResponse.json()
+
+    elif functionResponse.status_code == 202: # try once more
+        # 202 Accepted activation request (should not happen while sending 'blocking=true&result=true')
+        logger.warning("Did not receive response from function '%s' in package '%s', trying once more.", functionName, package)
+        logger.info(json.dumps(functionResponse.json()))
+        responseJson = functionResponse.json()
+        activationId = responseJson['activationId']
+        functionCallUrl = CFNamespaceUrl + '/activations/' + activationId + '/result'
+        logger.info("Trying to get function result from url '%s'", functionCallUrl)
+        functionResponse = requests.get(functionCallUrl, auth=(username, password),
+                                        headers={})
+        if functionResponse.status_code == 200:
+
+            responseContentType = functionResponse.headers.get('content-type')
+            if responseContentType != 'application/json':
+                logger.error('Response content type is not json, content type: %s, response:\n%s', responseContentType, functionResponse.text)
+                return None
+
+            responseJson = functionResponse.json()
+            if not 'result' in responseJson or not responseJson['result'] or not 'payload' in responseJson['result']:
+                logger.error("Bad response format received from function '%s' in package '%s', status code '%d',\
+                 response: %s, expected was {\"result\":{\"payload\":\"Hello, Fred!\"},\"success\":true,\"status\":\"success\"}",
+                 functionName, package, functionResponse.status_code, json.dumps(functionResponse.json(), ensure_ascii=False).encode('utf8'))
+                return None
+            else:
+                return responseJson['result']['payload']
+
+        elif functionResponse.status_code in [403, 404, 408]:
+            # 403 Forbidden (could be just for specific package or function)
+            # 404 Not Found (action or package could be incorrectly specified for given test)
+            # 408 Request Timeout (could happen e.g. for CF that calls some REST APIs, e.g. Discovery service)
+            # 502 Bad Gateway (when the CF raises exception, e.g. bad params where provided)
```

Comment:
Please move this comment to else branch below as this is not in the list of codes that match this branch.

Reply:
Moved.